### ISSUE TITLE
RTMP stream is not possible with new version

### ISSUE THREAD

**Issue by satyamub** [jgh-/VideoCore#288]
@jgh- This is third time I am requesting this issue.

If I use the 0.1.12 version of video core in cocoapod then, it starts streaming to Adobe FMS server however to use the latest features and to overcome some issue, if I use 0.3.2 then it does not even start the streaming and it throws the below log. Can you please let me know if there has any changes needs to be done at server end or is this an issue?

ClientState: 1
ClientState: 2
ClientState: 3
Want read:4096, read:3073
ClientState: 4
ClientState: 5
ClientState: 6
Tracking command(1, connect)
Want read:4096, read:336
Steam in buffer size:336
First byte:0x2, header type:0
Handle message:5
Received server window size: 2500000
Steam in buffer size:320
First byte:0x2, header type:0
Handle message:6
Received peer bandwidth limit: 2500000 type: 2
Steam in buffer size:303
First byte:0x2, header type:0
Handle message:1
Request to change incoming chunk size from 128 -> 4096
Steam in buffer size:287
First byte:0x3, header type:0
Handle message:20
Received invoke
Received invoke _result
Find command: connect for ID:1
Tracking command(2, releaseStream)
Tracking command(3, FCPublish)
Tracking command(4, createStream)
ClientState: 7
Steam in buffer size:33
First byte:0x3, header type:0
Handle message:20
Received invoke
Received invoke onBWDone
Want read:4096, read:0
Read from stream error:0
ClientState: 11

**Comment by satyamub** [comment]
@jgh-  can you please let me know the issue?

**Comment by satyamub** [comment]
@jeffreywescott @cloudwalking can you al please look into this issue and let  me know whats happening?

**Comment by satyamub** [comment]
@jgh- Hi James, I have been trying to contact you for this issue since Dec 2015. Please let me know if this is unworkable when updated to 0.3.2

**Comment by JALsnipe** [comment]
Spamming mentions and duplicating your issue won't help you get a response...

**Comment by satyamub** [comment]
@JALsnipe how to get notified then>?

**Comment by satyamub** [comment]
@JALsnipe @jgh- Thanks all for not helping me and I AM DONE....Resolved the issue.

@jgh- Thanks again for this Wonderful Library!!!

**Comment by chinnuios** [comment]
@satyamub  Hi I am  also facing same issue .I am working with Red5 server and videocore 0.1.12. But I am unable start the session every time. every 3 to 4 attempts it started working. can you please tell me this Library works with the Red5 or not.

**Comment by satyamub** [comment]
@chinnuios .. can you please let me know what exactly happens when you try to start the streaming? and also let me know the streaming server  address/stream name so that I will start streaming from here and check for it?

**Comment by chinnuios** [comment]
@satyamub Thanks for the replay . I am using Red5 server and if start the session it will disconnected after few sec i tried with videocore 0.1.12 but it is also not working perfectly.
1)will it work with red5 or not. If yes can you please provide your red5 server name then i will check on project.
2)which library your using for Player.

**Comment by JFKingsley** [comment]
@satyamub: I'm currently having the same issue. Could you explain the fix that you found?

**Comment by satyamub** [comment]
can you give me email address?

**Comment by JFKingsley** [comment]
Email's on my profile.

**Comment by chinnuios** [comment]
@satyamub       I am having error at    strncpy(propName, (char*)p, nameLen); . while i'm have to given streamKey: 123456789045/456872984165 . please help me

**Comment by satyamub** [comment]
@JFKingsley  which streaming server are you using?

**Comment by satyamub** [comment]
@chinnuios which has file has this line? can you please let me know?

**Comment by JFKingsley** [comment]
@satyamub All of them. I've tried it with Wowza, Red and Adobe, all with the same error.

**Comment by satyamub** [comment]
@JFKingsley I have sent an email. please check and let me know if that is working with your servers!!

**Comment by haemi** [comment]
should it work with YouTube rtmp as well?

**Comment by satyamub** [comment]
I don't know @haemi.. I have not tried.. you can try..

**Comment by haemi** [comment]
I do not know how the config JSON should look like...

**Comment by chinnuios** [comment]
@satyamub In RTMPSession.cpp file and see #302

**Comment by satyamub** [comment]
@chinnuios  what do you want me to see there?

**Comment by ntdang3003** [comment]
I have the same issue. My email in my profile.
@satyamub 👍  I have sent an email about this issue for you. Can you help me how to fix this issue?
@JFKingsley  👍  Do you solve the issue yet. If you do, can you help me :)

**Comment by satyamub** [comment]
@ntdang3003 I did not receive any email from you!!!

**Comment by ntdang3003** [comment]
@satyamub Sorry, it can be a problem with my email. Can you please send me an email to contact to you? Can you help me to fix that issue? My email is in my profile. Thank so much.

**Comment by JFKingsley** [comment]
Yeah.. I've tried pretty much every fix proposed to be, with none of them working. I'd say unless @jgh- can fix this, this project is considered unfortunately abandoned and non-functional.

**Comment by satyamub** [comment]
@JFKingsley can you send me the Adobe server details you have set up so that i could start and check for that issue because I have tested and have a working copy with adobe server!!

**Comment by chinnuios** [comment]
@satyamub  May i know your streamkey . How many character did you had ??

**Comment by pthomsen-bb** [comment]
@satyamub did you get streaming to Red5 to work? 

If so, what changes did you have to make? I'm in the same situation, and I'm getting disconnected from Red5 (1.0.6) whenever I try to connect from VideoCore 0.3.2.

Feel free to email me (address in the profile) if that is more convenient to you.

Thanks in advance!

**Comment by satyamub** [comment]
@pthomsen-bb can you test that same thing in Red5 1.0.4 and let me know the issue.

**Comment by pthomsen-bb** [comment]
Server logs for the relevant connection attempt.
[red5-1.0.4-log-snippet.txt](https://github.com/jgh-/VideoCore/files/250959/red5-1.0.4-log-snippet.txt)

Client log from Xcode:

2016-05-05 11:07:49.622 rtmp[616:222677] startStream
Connecting:xxxxxx.yyyyyyy.com:1935, stream name:pthomsen
ClientState: 1
ClientState: 2
ClientState: 3
2016-05-05 11:07:49.965 rtmp[616:222677] VCSessionStateStarting
2016-05-05 11:07:49.970 [info][tid:com.facebook.react.JavaScript] VCSessionStateStarting
Want read:4096, read:3073
ClientState: 4
ClientState: 5
ClientState: 6
Tracking command(1, connect)
Want read:4096, read:29
Steam in buffer size:29
First byte:0x2, header type:0
Handle message:5
Received server window size: 10000000
Steam in buffer size:13
First byte:0x42, header type:1
Handle message:6
Received peer bandwidth limit: 10000000 type: 2
Want read:4096, read:252
Steam in buffer size:252
First byte:0x42, header type:1
Handle message:4
Received ping, sending pong.
Steam in buffer size:238
First byte:0x3, header type:0
Handle message:20
Received invoke
Received invoke _result
Find command: connect for ID:1
Tracking command(2, releaseStream)
Tracking command(3, FCPublish)
send pong
Tracking command(4, createStream)
ClientState: 7
Want read:4096, read:0
Read from stream error:0
ClientState: 11
~RTMPSession2016-05-05 11:07:50.597 rtmp[616:222677] VCSessionStateEnded
2016-05-05 11:07:50.598 [info][tid:com.facebook.react.JavaScript] VCSessionStateEnded
2016-05-05 11:07:50.598 rtmp[616:222677] VCSessionStateEnded
2016-05-05 11:07:50.599 [info][tid:com.facebook.react.JavaScript] VCSessionStateEnded

**Comment by IOSDeveloperProgrammer** [comment]
@satyamub  Hi, Video core working perfectly for you ? I having some problem .Please help me

**Comment by satyamub** [comment]
@IOSDeveloperProgrammer  send me your email address? by the way which server you are using?

**Comment by IOSDeveloperProgrammer** [comment]
@satyamub HI, Thanks for given response. Problem is : Camera open , After  delay for 10 - 30sec when i was click on connect button .it's occurring by repeatedly click on  connect button.How to reduce delay camera ? Did you find any solutions ?? can you please help me on this issue.

**Comment by satyamub** [comment]
@IOSDeveloperProgrammer which device and which iOS version?

**Comment by IOSDeveloperProgrammer** [comment]
@satyamub  Iphone6, ios version: 9.3.2
